docs: MDCB 2.10.0 release notes (#1822)
TykTechnologies/tyk-docs: developer-support/release-notes/mdcb.mdx

---

## 2.10 Release Notes

### 2.10.0 Release Notes

#### Release Date xx April 2026

#### Release Highlights

> Tyk MDCB has been updated to Go 1.25 for enhanced security and performance. This release also addresses some CVEs and adds a new `omit_analytics_index_creation` configuration option to prevent PostgreSQL deadlocks when running multiple MDCB instances.

Reviewer: ```suggestion
Tyk MDCB has been updated to Go 1.25 for enhanced security and performance. This release also addresses some CVEs and exposes more controls to configure the transfer of aggregated traffic logs to the Control Plane data store.
```

| [MongoDB](https://www.mongodb.com/try/download/community)  | 5.0.x, 6.0.x, 7.0.x | 4.4.x, 5.0.x, 6.0.x, 7.0.x | |
| [PostgreSQL](https://www.postgresql.org/download/)         | 13.x - 17.x        | 13.x - 17.x            | |

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions
If you are upgrading to 2.10.0, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads
- [Docker image v2.10.0](https://hub.docker.com/r/tykio/tyk-mdcb-docker/tags?page=&page_size=&ordering=&name=v2.10.0)
  - ```bash
    docker pull tykio/tyk-mdcb-docker:v2.10.0
    ```
#### Changelog
<a id="Changelog-v2.10.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

<Accordion title='Updated Go version to 1.25'>
The Tyk MDCB has been updated to Golang 1.25, improving security by staying up-to-date with Go versions.
</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>

> <Accordion title='Added option to disable analytics index creation in horizontally scaled MDCB deployments'>

Reviewer: ```suggestion
<Accordion title='Extended existing MongoDB option to avoid database contention when using PostgreSQL data store in the Control Plane'>
```

> Added a new `omit_analytics_index_creation` configuration option to prevent PostgreSQL deadlocks in horizontally scaled MDCB deployments. When enabled, MDCB instances skip automatic index creation while maintaining full analytics functionality, eliminating database contention issues when multiple instances attempt to create the same indexes concurrently.

Reviewer: ```suggestion
The [`omit_analytics_index_creation`](/tyk-multi-data-centre/mdcb-configuration-options#omit_analytics_index_creation) configuration option is used to prevent deadlocks in horizontally scaled deployments that could occur if multiple MDCB instances created the same index concurrently.
When this option is enabled, MDCB does not [automatically create indexes](/api-management/tyk-pump#omitting-indexes), eliminating database contention issues, while maintaining full analytics functionality.
This control was previously only applied when targeting a MongoDB data store in the Control Plane. Now it can also be used with a PostgreSQL data store.
```

</Accordion>

> <Accordion title='Added support for `ignore_aggregations` configuration option'>

Reviewer: ```suggestion
<Accordion title='Added options to allow better control of high cardinality data when using MongoDB store in the Control Plane'>
```

> Added the ability to configure the `ignore_aggregations` option for the MongoDB aggregates pump. This allows users to better control high cardinality data, such as per-key analytics, which helps prevent performance degradation and reduces database storage costs.

Reviewer: ```suggestion
We have added additional configuration options when using a MongoDB data store in the Control Plane. These allow better control of high cardinality data, such as per-key analytics, which helps prevent performance degradation and reduces database storage costs.

- [`ignore_aggregations`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_ignore_aggregations) accepts a list of aggregations that should not be passed to the Control Plane
- [`enable_aggregate_self_healing`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_enable_aggregate_self_healing) automatically creates a new document in MongoDB when the maximum document size is reached.

```

</Accordion>

> <Accordion title='Added support for self-healing aggregate pump'>

Reviewer: ```suggestion
```

> Added the ability to enable the `enable_aggregate_self_healing` option for the MongoDB aggregates pump. When enabled, this feature automatically splits aggregate documents that exceed database size limits, preventing analytics data loss and reducing network and compute resource consumption.

Reviewer: ```suggestion
```

---

## 2.10 Release Notes

### 2.10.0 Release Notes

> #### Release Date xx April 2026

Reviewer 2: ```suggestion
#### Release Date 23 April 2026
```

</Accordion>

<Accordion title='Added options to allow better control of high cardinality data when using MongoDB store in the Control Plane'>
We have added additional configuration options when using a MongoDB data store in the Control Plane. These allow better control of high cardinality data, such as per-key analytics, which helps prevent performance degradation and reduces database storage costs.

- [`ignore_aggregations`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_ignore_aggregations) accepts a list of aggregations that should not be passed to the Control Plane
- [`enable_aggregate_self_healing`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_enable_aggregate_self_healing) automatically creates a new document in MongoDB when the maximum document size is reached.
</Accordion>

</AccordionGroup>

##### Security Fixes

<AccordionGroup>

<Accordion title='CVE fixed'>
Addressed the following CVEs, providing increased protection against security
vulnerabilities, including, but not limited to:

> - <a href="https://nvd.nist.gov/vuln/detail/CVE-2026-33186" target="_blank">CVE-2026-33186</a

Reviewer 2: ```suggestion
- <a href="https://nvd.nist.gov/vuln/detail/CVE-2026-33186" target="_blank">CVE-2026-33186</a>
```

When this option is enabled, MDCB does not [automatically create indexes](/api-management/tyk-pump#omitting-indexes), eliminating database contention issues, while maintaining full analytics functionality.
This control was previously only applied when targeting a MongoDB data store in the Control Plane. Now it can also be used with a PostgreSQL data store.
</Accordion>

<Accordion title='Added options to allow better control of high cardinality data when using MongoDB store in the Control Plane'>
We have added additional configuration options when using a MongoDB data store in the Control Plane. These allow better control of high cardinality data, such as per-key analytics, which helps prevent performance degradation and reduces database storage costs.

- [`ignore_aggregations`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_ignore_aggregations) accepts a list of aggregations that should not be passed to the Control Plane
- [`enable_aggregate_self_healing`](/tyk-multi-data-centre/mdcb-configuration-options#analytics-mongo_enable_aggregate_self_healing) automatically creates a new document in MongoDB when the maximum document size is reached.

> 

Reviewer: ```suggestion
```

</Accordion>

> 

Reviewer: ```suggestion
```

> </Accordion>

Reviewer: ```suggestion
```
